keyboards/preonic/keymaps/via/keymap.c

```diff
@@ -0,0 +1,310 @@
+/* Copyright 2015-2017 Jack Humbert
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "muse.h"
+
+enum preonic_layers {
+  _QWERTY,
+  _COLEMAK,
+  _DVORAK,
+  _LOWER,
+  _RAISE,
+  _ADJUST
+};
+
+enum preonic_keycodes {
+  QWERTY = SAFE_RANGE,
+  COLEMAK,
+  DVORAK,
+  LOWER,
+  RAISE,
+  BACKLIT
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```

Comment:
Via only supports 4 layers by default.  While this can be increased, the board actually has to have enough eeprom to do so. 

The simplest solution here is to reduce the number of layers to 4, base, raise, lower, and adjust. 

Also there are the `KC_FN13` and `KC_FN23` keycodes that perform the tri-layer stuff.

Reply:
Thanks @drashna! I have found that both the Planck and Preonic can both handle all 6 layers in VIA well, but there are some issues with the 6th layer keymap - via doesn't recognize key codes like DF yet... so it all comes out as hex codes. they still work, but its not easily human readable which seems the point of VIA.

Anyhow, I can reduce it to 4 layers if you all think that is better, but it hasn't been an issue.

I am not sure I understand about `KC_FN13` and `KC_FN23`?

Reply:
I think reducing it 4 layers may be best, but I'd really say that @jackhumbert should have the final say on that. 

And the keycodes listed do the tri-layer stuff. Oh, sorry, they're `FN_MO13` and `FN_MO23`.
https://github.com/qmk/qmk_firmware/blob/135d28d2fb3234dab3ef5353b47ec5838534e39d/quantum/via.c#L163-L183

Reply:
Sorry for the delay. I reduced to 4 layers as suggested, changed RAISE and LOWER to FN_MO13 and FN_MO23 respectively, and removed some the additional keycodes not understood by VIA. Thank you for the advice and suggestions!

I will implement some of these in my Planck PR when i get my replacement PCB from OLKB group buy.

All seem to be working on my end, so just waiting for approval from @jackhumbert and a good VID for OLKB :)

Reply:
@mixedfeelings how do i go about testing this via support on my Preonic?

Reply:
@josuemldnd23  are you able to clone my branch and compile the firmware? otherwise i can send you what i have compiled.

Reply:
@mixedfeelings would also love to test if I can. I'm new to gmk, so I am looking into compiling this, but if you have something already, I would take that too. ;)

Reply:
@josuemldnd23 @natedunn here is the firmware i am using: 
https://github.com/mixedfeelings/VIA-layouts/blob/main/OLKB/Preonic/rev3/firmware/preonic_rev3_via.bin

along with this VIA JSON:
https://github.com/mixedfeelings/keyboards/blob/preonic-rev3/src/olkb/preonic/rev3/olkb-preonic-rev3.json